src/nvim/mapping.c

```diff
@@ -2118,6 +2130,7 @@ static Dictionary mapblock_fill_dict(const mapblock_T *const mp, const char *lhs
   PUT_C(dict, "mode", CSTR_AS_OBJ(mapmode));
   PUT_C(dict, "abbr", INTEGER_OBJ(abbr ? 1 : 0));
   PUT_C(dict, "mode_bits", INTEGER_OBJ(mp->m_mode));
+  PUT_C(dict, "default", BOOLEAN_OBJ(mp->m_default));
```

Comment:
It may be a bit inconsistent for this to be a boolean as there are no other boolean fields. For consistency this should be an integer, but that's inconvenient for Lua plugins. Since this is mostly likely to be used by Lua plugins, maybe it should be like this:
```suggestion
  if (mp->m_default) {
    PUT_C(dict, "default", INTEGER_OBJ(1));
  }
```
In future it may also be possible for the `default` field to have a different type (e.g. string) to provide more information, in which case a missing `default` field indicating a non-default mapping makes more sense.

Reply:
Changed as suggested d72a4beebea0c6d658d03bdcd8352ccdfcde2611.